eyes/eyes.py

```diff
@@ -0,0 +1,379 @@
+
+    """
+    def can_bell(self, channel_or_server):
+        cs = channel_or_server
+        bells = self.settings['BELLS']
+        return self.bell_id(channel_or_server) in bells"""
+
+    def bell_id(self, channel_or_server):
+        cs = channel_or_server
+        if isinstance(cs, discord.Server):
+            return 'S' + cs.id
+        return cs.id
+
+    def save(self):
+        return dataIO.save_json(SETTINGS_PATH, self.settings)
+
+    def highlight_me(self, content, server):
+        resume = C.ENDC
+        if content.startswith(C.WARNING):
+            resume = C.WARNING
+        if content.startswith(C.FAIL):
+            resume = C.FAIL
+        return re.sub(self.settings['BELL_PATTERN'],
+                      C.ENDC + C.BELL + C.OKBLUE + r'\g<0>' + resume,
+                      content)
+
+    async def respond(self, yes):
+        if yes:
+            return await self.bot.say(":eyes:")
+        m = await self.bot.say(":hear_no_evil:")
+        await asyncio.sleep(.5)
+        await self.bot.edit_message(m, ":see_no_evil:")
+
+    @checks.is_owner()
+    @commands.group(pass_context=True)
+    async def eyes(self, ctx):
+        """settings for your 👀"""
+        if ctx.invoked_subcommand is None:
+            await send_cmd_help(ctx)
+
+    @eyes.command(pass_context=True, name="server", no_pm=True)
+    async def eyes_server(self, ctx):
+        """you can't hide 👀"""
+        server = ctx.message.server
+        serv_sets = self.settings['SERVERS']
+        chan_sets = self.settings['CHANNELS']
+        try:  # lol
+            serv_sets.remove(server.id)
+        except ValueError:
+            serv_sets.append(server.id)
+        else:  # more natural
+            chan_sets = [cid for cid in chan_sets
+                         if not server.get_channel(cid)]
+        self.save()
+        await self.respond(server.id in serv_sets)
+
+    @eyes.command(pass_context=True, name="channel", no_pm=True)
+    async def eyes_channel(self, ctx, channel: discord.Channel=None):
+        """channel 👀
+        defaults to current"""
+        channel = channel or ctx.message.channel
+        if channel.type is not discord.ChannelType.text:
+            ugh = randchoice(":neutral_face: :confused: :expressionless: "
+                             ":unamused: :rolling_eyes: :thinking:".split())
+            await self.bot.say(ugh)
+            return
+        chan_sets = self.settings['CHANNELS']
+        try:
+            chan_sets.remove(channel.id)
+        except ValueError:
+            chan_sets.append(channel.id)
+        self.save()
+        await self.respond(channel.id in chan_sets)
+
+    @eyes.command(pass_context=True, name="list", no_pm=True)
+    async def eyes_list(self, ctx):
+        """list 👀s in server"""
+        server        = ctx.message.server
+        me            = server.me
+        text_channels = [c for c in server.channels
+                         if c.type is discord.ChannelType.text]
```

Comment:
oops, don't check for read_permission here. should we do so when adding?
mmm naw.
but should we let the user know somehow if a channel is not readable?